### ISSUE TITLE
GET /bundles/all ignores invalid prefix

### ISSUE THREAD

**Issue by jessebrennan** [HumanCellAtlas/data-store#2643]
```
jesse@vader ~$ http -b get "https://dss.data.humancellatlas.org/v1/bundles/all?replica=aws&prefix=F"
{
    "bundles": [],
    "dss_api": "https://dss.data.humancellatlas.org",
    "event_timestamp": "2019-12-02T214901.570541Z",
    "has_more": false,
    "object": "list",
    "page_count": 0,
    "per_page": 100,
    "search_prefix": "bundles/F"
}
```
but
```
jesse@vader ~$ http -b get "https://dss.data.humancellatlas.org/v1/bundles/all?replica=aws&prefix=f"
{
    "bundles": [
...
    ],
    "dss_api": "https://dss.data.humancellatlas.org",
    "event_timestamp": "2019-12-02T214929.423574Z",
    "has_more": true,
    "link": "https://dss.data.humancellatlas.org/v1/bundles/all?prefix=f&replica=aws&search_after=bundles%2Ff02cae0f-7cf3-474e-aa8b-cf0e09de7518.2019-08-01T200147.791889Z&token=1T0Nj%2FkyWssgaGbl%2FKM%2FfIrVMypR6Szwy3i0q5zkHckkrsAXMh2KJ92k2k9EtIHIbvNYLXeUNiiVc17aEj%2FE%2BUv%2F1tgZU9ST5wI4jrMEIC0x15e8PfDaF2VAHSrHqnA6pcUAx%2BgVt3Pw%3D",
    "object": "list",
    "page_count": 100,
    "per_page": 100,
    "search_after": "bundles/f02cae0f-7cf3-474e-aa8b-cf0e09de7518.2019-08-01T200147.791889Z",
    "search_prefix": "bundles/f",
    "token": "1T0Nj/kyWssgaGbl/KM/fIrVMypR6Szwy3i0q5zkHckkrsAXMh2KJ92k2k9EtIHIbvNYLXeUNiiVc17aEj/E+Uv/1tgZU9ST5wI4jrMEIC0x15e8PfDaF2VAHSrHqnA6pcUAx+gVt3Pw="
}

```
This isn't necessarily a bug, but I think there is room for improvement. Either
- the OpenAPI spec should warn that invalid prefixes will be ignored, or
- there should be some validation on the query parameter that lets the user know if the prefix is not one of the expected values ([a-f0-9][a-f0-9\\-]*).

Whether the DSS should ignore case for the prefix is a separate, but relevant question. I think it would be helpful to list the decided answer in the OpenAPI spec.

**Comment by DailyDreaming** [comment]
We lowercase when a user submits a uuid with captial letters.  I think it's reasonable that if the user then searches for that same uuid, we again lower-case the prefix for them.

**Comment by DailyDreaming** [comment]
Closed by #2649.

**Comment by chmreid** [comment]
Reopening so I can connect this issue to #2664 via zenhub
